library/src/commonMain/kotlin/com/eygraber/sqldelight/androidx/driver/AndroidxSqliteDriver.kt

```diff
@@ -426,6 +456,52 @@ public class AndroidxSqliteDriver(
   }
 }
 
+private inline fun SQLiteConnection.withDeferredForeignKeyChecks(
+  configuration: AndroidxSqliteConfiguration,
+  block: () -> Unit,
+) {
+  if(configuration.isForeignKeyConstraintsEnabled) {
+    prepare("PRAGMA foreign_keys = OFF;").use(SQLiteStatement::step)
+  }
+
+  block()
+
+  if(configuration.isForeignKeyConstraintsEnabled) {
+    prepare("PRAGMA foreign_keys = ON;").use(SQLiteStatement::step)
+
+    if(configuration.isForeignKeyConstraintsCheckedAfterCreateOrUpdate) {
+      prepare("PRAGMA foreign_key_check;").use { check ->
+        val violations = mutableListOf<String>()
+        while(check.step()) {
+          val referencingTable = check.getText(0)
+          val referencingRowId = check.getInt(1)
+          val referencedTable = check.getText(2)
+          val referencingConstraintIndex = check.getInt(3)
+
+          violations.add(
```

Comment:
This seems to be creating one such strong for each row. For a large database, this can be hundreds of thousands of rows. We should limit the amount of items this will return

Reply:
It should be every row that has a constraint violation, which could be helpful in some scenarios, but I could add a flag to limit how much is returned.

Reply:
Yes, I was saying:  A wrong migration will for example break all constraints in a table. That number is 50+k for many of my customers.  I dont need to know r very single one to understand something is wrong. I suggest just simply limiting it to ~100 rows + total count at most.

Reply:
True, but I don't want to be in a scenario where someone needed the 101st row to find the problem. I'll update it to not pass all of them to the exception's message, expose the violations as a property, and add a flag to limit how many results are read.

Reply:
I thought about it more and went with a default of 100 - https://github.com/eygraber/sqldelight-androidx-driver/pull/113